src/density.js

```diff
@@ -139,6 +156,12 @@ export default function() {
     return arguments.length ? (threshold = typeof _ === "function" ? _ : Array.isArray(_) ? constant(slice.call(_)) : constant(_), density) : threshold;
   };
 
+  density.aggregation = function(_) {
+    if(!arguments.length) return aggregation;
+    if(_ !== 'SUM' && _ !== 'MEAN') throw new Error("invalid aggregation");
```

Comment:
Could this be passed as a reducer function (in JavaScript)? What arguments would it take?

(Also it's a cosmetic detail but I don't think we'd use these ALLCAPS keywords.)

Reply:
It definitely could. In order to pass a list of the values per grid cell, we would need to store all of those. That could have a performance implication, in contrast to the current incremental computation which only requires a constant amount (1 previously, 2 in this PR) of values per grid cell (with the `+=` operations).

Would you be open to implementation of 
```typescript
.aggregation(strategy : 'sum' | 'min' | 'max' | 'mean' | (cellValues: number[]) => number)
```

where the per-cell values are only stored if the user passes the custom reducer? It would certainly make the code more complex, but I am currently using this with over 300,000 points (and millions in the future), so I would want the option to avoid the overhead of storing the per-cell data for the most common aggregation strategies.